worker/pruner/worker.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package pruner
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/watcher"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.pruner")
+
+// Facade represents an API that implements status history pruning.
+type Facade interface {
+	Prune(time.Duration, int) error
+	WatchForModelConfigChanges() (watcher.NotifyWatcher, error)
+	ModelConfig() (*config.Config, error)
+}
+
+//updates model configuration for any implementation of a pruner worker
+type ModelConfigUpdater interface {
+	MaxEntryAgeUpdate() time.Duration
```

Comment:
Ok, now I understand what this is doing. I think it would be nicer if  `PrunerWorker.Work` took a `PrunerConfigFunc`:
```
type PrunerConfigFunc func(*config.Config) (time.Duration, uint)
```
So the func passed to the action pruner would return the action max age and size, and the one passed to the status history pruner would return the age and size for status history.